### PR TITLE
New modules: iam_instance_profile(_info)

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

New modules for listing/managing IAM Instance Profiles

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

iam_instance_profile
iam_instance_profile_info

##### ADDITIONAL INFORMATION

Fixes: https://github.com/ansible-collections/community.aws/issues/1842

### CODE REVIEW

**Comment by alinabuzachis** [comment]
recheck

File: plugins/modules/iam_instance_profile_info.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    prefile: /some/path/
+  register: result
+"""
+
+RETURN = r"""
+iam_instance_profiles:
+  description: List of IAM instance profiles
+  returned: always
+  type: complex
+  contains:
```
[the remark this answers is not in the document]
Comment (tremble) [reply plugins/modules/iam_instance_profile_info.py:58]: "sometimes".  The AWS APIs were being inconsistent, so the code handles them but we don't guarantee them :)

**Review by alinabuzachis — APPROVED** [review]

**Comment by tremble** [comment]
regate

**Comment by tremble** [comment]
regate

**Pull request merged**